src/olympia/files/utils.py

```diff
@@ -742,16 +742,12 @@ def __init__(self, source, mode='r', force_fsync=False):
         self.info_list = None
         self.mode = mode
         self.force_fsync = force_fsync
-        self.is_valid = self.initialize_and_validate()
+        self.initialize_and_validate()
 
     def initialize_and_validate(self):
         """
         Runs some overall archive checks.
         """
-        # Shortcut to avoid expensive check over and over again
-        if getattr(self, 'is_valid', False):
```

Comment:
Why did you remove this optimization? Did it never get used?

Reply:
It wasn't - if you look at the whole patch, there is only one place where we used `is_valid` left (that I removed), and even there it was dead code, because an exception would have be thrown before we got to that point.

Reply:
Right, but wasn't the idea of this that if we had to call `initialize_and_validate` on the same object more than once we could shortcut it? I assume there's nowhere in the code where `initialize_and_validate` might be called more than once on the same object?

Reply:
It's called at `__init__` time so there is no way around it (I think historically it wasn't always the case, and that's why this property made sense)